### PR TITLE
[WIP] Refactoring tests to support multiple backends

### PR DESCRIPTION
**Pull request by dantleech**
In order to support the ORM the event listener tests must be abstracted.

### CODE REVIEW

**Comment by dantleech** [comment]
Have rewritten this PR.

It is now reorganized so that we can start making the tests generic.

- There is a test "repository" for each adapter implementation.
- The repository is chosen depending on the environment.
- The "adapter/listener" tests use the repository.

**Comment by dantleech** [comment]
Most of the tests are still PHPCR-ODM specific, the idea is to start moving them into the abstract class as the (to be introduced) ORM adapter supports them.

File: Tests/Resources/app/AppKernel.php [hunk 1]
```diff
@@ -21,6 +21,17 @@ public function configure()
...
+        $loader->import(CMF_TEST_CONFIG_DIR.'/default.php');
+        $loader->import(CMF_TEST_CONFIG_DIR.'/phpcr_odm.php');
+        $loader->import(__DIR__ . '/config/' . $this->environment . '.yml');
+    }
+
+
+    protected function buildContainer()
+    {
+        $container = parent::buildContainer();
+        $container->setParameter('cmf_testing.bundle_fqn', 'Symfony\Cmf\Bundle\RoutingAutoBundle');
```
Comment (dantleech) [inline Tests/Resources/app/AppKernel.php]: We should really get rid of the above "magic"

**Comment by almacbe** [comment]
I think this PR we could close it because we have added support to ORM and the test of this part is on the other bundle.

**Comment by dbu** [comment]
agreed. thanks for your work @dantleech and @dantleech

**Pull request closed without merging**